### PR TITLE
Do not notify bytecode release for debugging-unmarked function

### PR DESCRIPTION
**Pull request by clover2123**
Signed-off-by: HyukWoo Park <hyukwoo.park@samsung.com>

### CODE REVIEW

**Comment by clover2123** [comment]
@zherczeg 
Recently, I have updated the debugger not to generate debugging-bytecode(breakpoint) for unmarked code such as dynamically create functions (new Function()).
Every `ByteCodeBlock` invokes the finalizer when it is deallocated through gc, and this finalizer calls `clearByteCodeBlock` below.
https://github.com/Samsung/escargot/blob/21d0e6aa42ffcf8a788395c08b1dbc6de44bba7a/src/interpreter/ByteCode.cpp#L118-L125

I'm wondering that if a ByteCodeBlock which has no debugging code (debugging-unmarked) is deallocated and then its finalizer is called, would notification of bytecode-release incur any conflict?

File: src/interpreter/ByteCode.cpp [hunk 1]
```diff
@@ -119,7 +119,7 @@ static void clearByteCodeBlock(ByteCodeBlock* self)
 {
 #ifdef ESCARGOT_DEBUGGER
     Debugger* debugger = self->m_codeBlock->context()->debugger();
-    if (debugger != nullptr) {
+    if (debugger != nullptr && self->codeBlock()->markDebugging()) {
```
Comment (bbrto21) [inline src/interpreter/ByteCode.cpp:122]: `self->m_codeBlock` vs `self->codeBlock()`
Which one should be preferred?

**Comment by zherczeg** [comment]
The point of this operation is that the debugger might (and do) have extra data associated for the byte code, and this call allows it to clean up this data. If a byte code is not used (knwon) by the debugger, this call should be unnecessary, so your code is correct.

File: src/interpreter/ByteCode.cpp (hunk 1 above)
Comment (clover2123) [reply src/interpreter/ByteCode.cpp]: Good catch! I fixed it using `codeBlock()` method

**Comment by clover2123** [comment]
@zherczeg Thanks for your detailed explanation :)

**Review by zherczeg — APPROVED** [review]
LGTM (informal)

**Pull request merged**